### ISSUE TITLE
2 last characters missing

### ISSUE THREAD

**Issue by burczyk** [NimbusKit/markdown#6]
I cloned repo from @myell0w
https://github.com/jverkoey/NSAttributedStringMarkdownParser/commit/4e649d43b089036405761db94cc00d50b322e7e9
and I saw that after parsing markdown utf-8 text 2 last characters are missing. 
Does anyone have the same problem?

**Comment by oemera** [comment]
I have the same issue with utf-8 characters like 'ü', 'ö' and so on! Is there a way to fix this?

**Comment by dimitry** [comment]
Same issue here. Looks like string length is incorrectly computed when dealing with unicode characters.

**Comment by burczyk** [comment]
I 'hacked' it in my app by counting non-standard characters and adding spaces in the end of string. Not pretty, but worked.

**Comment by dimitry** [comment]
@burczyk I'm completely content to do that at this point :) Project needs to launch. Any code snipped to help me get there as well? Are you just counting how many chars are 128 and above? Thank you.

**Comment by burczyk** [comment]
@dimitry yes, something like that. Here you have a few ways to implement it:
http://stackoverflow.com/questions/4191168/how-can-i-count-the-number-of-non-alphanumeric-characters-in-an-nsstring-object

**Comment by dimitry** [comment]
@burczyk spot on! Thank you for your help. And thank you @myell0w for UTF8 support.

**Comment by oemera** [comment]
It would be awesome if you could create a sample repository with your fix so other can use it. Thanks guys!

**Comment by dimitry** [comment]
I used @myell0w's fork and then added blank spaces to the end of the markdown string (before sending it to the parser). Here's a quick copy & paste:

``` obj-c
NSString *markdown = @"...."; // Your markdown string

// Count non-standard characters
NSUInteger length = [markdown length];
NSUInteger totalNonStandardCharacters = 0;
for(int i=0; i<length; i++) {
    if ([markdown characterAtIndex:i] > 127) {
        totalNonStandardCharacters++;
    }
}

// Make string with spaces (to pad the original text)
NSMutableString *padding = [[NSMutableString alloc] init];
for(int i=0; i<totalNonStandardCharacters; i++) {
    [padding appendString:@" "];
}

// Add padded spaces to the end of the string
markdown = [NSString stringWithFormat:@"%@%@", markdown, padding];

// Now send it to the parser
```

I think there's a cleaner way to create an empty string with X number of spaces, but no time for it right now. Quick & dirty!

**Comment by AlexDenisov** [comment]
Hi guys, here is a fix for this issue #12.
But I faced with another problem, as described here (#2) non-ASCII characters missed...
